dask/dataframe/core.py

```diff
@@ -1494,7 +1494,7 @@ def sem(self, axis=None, skipna=None, ddof=1, split_every=False):
                 result.divisions = (min(self.columns), max(self.columns))
             return result
 
-    def quantile(self, q=0.5, axis=0):
+    def quantile(self, q=0.5, axis=0, use_tdigest=True):
```

Comment:
Since we may add more methods in the future, I'd prefer this to be a string. Current values would be:

- `method='tdigest': uses tdigest
- `method='dask': uses dask's custom algorithm
- `method='default'`: the default value for this kwarg just means use the default for this version of dask. We should feel free to change this if we find a new best method. For now this would map to `'dask'`, but could be changed later.

Reply:
Implemented this change as described here. Although I must raise the concern here that defaulting to `'dask'` can lead to some pretty nasty surprises. For example in the case that brought this issue to my knowledge I wanted to get the 95th percentile from data and the call `ddf.quantile(0.95)` was giving me a value of ~12. A week later I learned about the issues with the quantile method and ran it using t-digest which game me a result of ~8. The internal implementation sometimes fails massively and it can be very difficult to spot. I think it's dangerous to use it as the default without a warning. I'm not sure what would be the best way to handle it without breaking existing code as t-digest can't handle all the cases the internal one can.

Reply:
@jcrist what are your thoughts on what the default should be?  Should we make crick mandatory for dask dataframe?

Reply:
Maybe? My thought was we should add the implementation's in this PR, let them sit for a bit so they can get some use, then maybe pull the switch and change the default. I think we do still want the current algorithm for partitioning logic (no external dependency, handles more dtypes), but for numerical results `tdigest` is likely better.

Reply:
Yeah I agree we still want to use the custom algorithm for handling partitioning. I'm fine with having it as `'dask'` for now, just wanted to make sure I bring that up since it was a pretty nasty surprise for me when I was starting to use Dask.